File: subprojects/frida-python/examples/open_service/xpc/listprocesses.py
import pprint

import frida

device = frida.get_usb_device()

appservice = device.open_service("xpc:com.apple.coredevice.appservice")
response = appservice.request(
    {
        "CoreDevice.featureIdentifier": "com.apple.coredevice.feature.listprocesses",
        "CoreDevice.action": {},
        "CoreDevice.input": {},
    }
)
pprint.pp(response)

File: subprojects/frida-node/examples/open_service/xpc/listprocesses.ts
import frida from "frida";
import { inspect } from "util";

async function main() {
    const device = await frida.getUsbDevice();

    const appservice = await device.openService("xpc:com.apple.coredevice.appservice");
    const response = await appservice.request({
        "CoreDevice.featureIdentifier": "com.apple.coredevice.feature.listprocesses",
        "CoreDevice.action": {},
        "CoreDevice.input": {},
    });
    console.log(inspect(response, {
        colors: true,
        depth: Infinity,
        maxArrayLength: Infinity
    }));
}

main()
    .catch(e => {
        console.error(e);
    });
